app/socket-apis/__tests__/get-user-post-ranks-and-top-ranked-whys.js

```diff
@@ -0,0 +1,128 @@
+// https://github.com/EnCiv/civil-pursuit/issues/208
+
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import { Mongo } from '@enciv/mongo-collections';
+import { ObjectId } from 'mongodb';
+import Rankings from '../../models/rankings';
+import Points from '../../models/points';
+import getUserPostRanksAndTopRankedWhys from '../get-user-post-ranks-and-top-ranked-whys';
+
+let memoryServer;
+let db;
+
+const synuser = { synuser: { id: new ObjectId().toString() } };
+
+beforeAll(async () => {
+  memoryServer = await MongoMemoryServer.create();
+  const uri = memoryServer.getUri();
+  await Mongo.connect(uri);
+  db = Mongo.db;
+  Rankings.setCollectionProps();
+  Points.setCollectionProps();
+});
+
+afterAll(async () => {
+  await Mongo.disconnect();
+  await memoryServer.stop();
+});
+
+beforeEach(async () => {
+  jest.spyOn(console, 'error').mockImplementation(() => {});
+});
+
+afterEach(async () => {
+  console.error.mockRestore();
+});
+
+test('User not logged in', async () => {
+  const callback = jest.fn();
+  await getUserPostRanksAndTopRankedWhys.call({}, 'discussion1', 1, ['id1', 'id2'], callback);
+  expect(callback).toHaveBeenCalledWith(undefined);
+  expect(console.error).toHaveBeenCalledWith(expect.stringMatching(/no user logged in/));
+});
+
+test('10 points in ids, nothing ranked', async () => {
```

Comment:
Instead of making new ids here, every time the test is run, generate 10 (use console.info) and then copy them into an array here.   

I'm learning that in the expect statements, what's expected needs to be more visible, meaning .objectToMatch({...}) so that when we look at the code in the test we can see what's expected.   -- We are missing things by not looking at it.